exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinMemoryCalculatorImpl.java

```diff
@@ -365,17 +299,20 @@ protected void initialize(boolean autoTune,
                               double loadFactor) {
       Preconditions.checkState(!firstInitialized);
       Preconditions.checkArgument(initialPartitions >= 1);
+      // If we had probe data before there should still be probe data now.
+      // If we didn't have probe data before we could get some new data now.
+      Preconditions.checkState(probeSizePredictor.hasData() && !probeEmpty || !probeSizePredictor.hasData());
```

Comment:
A simpler rewrite of the condition:  **! (probeEmpty && ! probeSizePredictor.hasData())**

Reply:
Done